pkg/controllerhelpers/scylla.go

```diff
@@ -296,6 +296,10 @@ func GetRackNodeCount(sdc *scyllav1alpha1.ScyllaDBDatacenter, rackName string) (
 }
 
 func IsScyllaDBDatacenterRolledOut(sdc *scyllav1alpha1.ScyllaDBDatacenter) (bool, error) {
+	if sdc == nil || len(sdc.Status.Conditions) == 0 {
+		return false, nil
+	}
+
 	if !helpers.IsStatusConditionPresentAndTrue(sdc.Status.Conditions, scyllav1alpha1.AvailableCondition, sdc.Generation) {
```

Comment:
isn't this used incorrectly then? The negation should be equivalent to `is status condition unknown or false` - it seems like this should match for an empty slice? Shall we unit test and fix this instead?

Reply:
negation of it is `is status condition not present or false`, either way, for both cases it correctly returns true for empty slice as condition is not present.